ir/memory.cpp

```diff
@@ -321,11 +321,13 @@ expr Pointer::is_block_alive() const {
 }
 
 expr Pointer::is_at_heap() const {
-  return m.blocks_kind.load(get_bid()) == 1;
+  // TODO: use bitvector instead of UF if there are few allocations
+  return get_value_from_uf("blks_at_heap", false, LOCAL_BLOCKS);
```

Comment:
non-local blocks may also be heap allocated. e.g. a pointer received by argument might be a malloc'ed ptr.

Reply:
You're right, fixed